### PR TITLE
Added utility functions to convert between msg::time and std::chrono time_point

### PR DESCRIPTION
**Pull request by ahcorde**
This PR is related with this issue https://github.com/ignitionrobotics/ign-common/issues/61. The idea it's to deprecate the class `common::Time` in favor of `std::chrono`.

Related with 
 - https://github.com/ignitionrobotics/ign-common/pull/90
 - https://github.com/ignitionrobotics/ign-sensors/pull/41
 - https://github.com/ignitionrobotics/ign-common/pull/90
 - https://github.com/ignitionrobotics/ign-math/pull/150

Added helpers functions to convert between
 - msg::Time to time_point
 - time_point to msg::Time
 - some tests too

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: src/Utility.cc
Comment (scpeters) [inline src/Utility.cc:212]: does the following work? It seems like it would have less chance for loss of precision:

~~~
auto duration = std::chrono::seconds(_time.sec()) + std::chrono::nanoseconds(_time.nsec());
~~~

Comment (scpeters) [reply src/Utility.cc:212]: we could also use `ignition::math::secNsecToTimePoint` from https://github.com/ignitionrobotics/ign-math/pull/150

if we don't want to wait for that to land, then I would add a TODO

File: src/Utility_TEST.cc
Comment (chapulina) [inline src/Utility_TEST.cc:322]: Style :wink: 

```suggestion
  std::chrono::steady_clock::time_point timePoint =
```

File: src/Utility.cc
Comment (chapulina) [inline src/Utility.cc:349]:
```suggestion
    msgs::Time Convert(const std::chrono::steady_clock::time_point &_timePoint)
```

**Review by chapulina — COMMENTED** [review]
We should also be sure to update the required `ign-math` version to 6.5.0 on `CMakeLists`. That's the same for all libraries.

**Review by chapulina — APPROVED** [review]
Pushed style updates and bumped the required `ign-math` version on 25e4f30, LGTM.

**Comment by chapulina** [comment]
@osrf-jenkins run tests now that the Jenkins server has been restarted

**Comment by chapulina** [comment]
Queued a new homebrew build now that `ign-math` 6.5.0 is in: https://build.osrfoundation.org/job/ignition_msgs-ci-pr_any-homebrew-amd64/

**Pull request merged**
